xbmc/video/VideoReferenceClock.cpp

```diff
@@ -94,18 +94,28 @@ void CVideoReferenceClock::Process()
   bool SetupSuccess = false;
   int64_t Now;
 
+  bool glx_fallback = false;
   while(!m_bStop)
   {
     //set up the vblank clock
 #if defined(HAVE_X11)
   std::string gpuvendor = g_Windowing.GetRenderVendor();
   std::transform(gpuvendor.begin(), gpuvendor.end(), gpuvendor.begin(), ::tolower);
-  if ((gpuvendor.compare(0, 5, "intel") == 0 ||
+  if (!glx_fallback && (gpuvendor.compare(0, 5, "intel") == 0 ||
        gpuvendor.compare(0, 5, "x.org") == 0)) // AMD
+  {
     m_pVideoSync = new CVideoSyncDRM();
 #if defined(HAS_GLX)
+    glx_fallback = true;
+#endif
```

Comment:
that if def is not needed. if you set glx_fallback to true and there is no glx, it will exit the loop on next cycle